src/interfaces/ast/map-ast.schema.json

```diff
@@ -1111,8 +1116,106 @@
             ],
             "type": "object"
         },
+        "MapDocumentNode": {
+            "additionalProperties": false,
+            "description": "Map document:\n`<header>`\n`<...definitions>`",
+            "properties": {
+                "astMetadata": {
+                    "$ref": "AstMetadata"
+                },
+                "definitions": {
+                    "items": {
+                        "anyOf": [
+                            {
+                                "$ref": "#/definitions/MapDefinitionNode"
+                            },
+                            {
+                                "$ref": "#/definitions/OperationDefinitionNode"
+                            }
+                        ]
+                    },
+                    "type": "array"
+                },
+                "header": {
+                    "$ref": "#/definitions/MapHeaderNode"
+                },
+                "kind": {
+                    "enum": [
+                        "MapDocument"
+                    ],
+                    "type": "string"
+                },
+                "location": {
```

Comment:
This is repeated quite often, don't we want to move it into its own definition and reference it with `$ref` field, if it's possible?

Reply:
i think it may be possible, but it's generated, so no big deal :tm: but i'll try

Reply:
it breaks, because ajv complains it has multiple Location references and i don't think it's worth fixing anyway :shrug: